tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp

```diff
@@ -118,4 +121,162 @@ TEST_F(VersionedLayerClientTest, GetPartitions) {
   EXPECT_TRUE(mock_server_client_->Verify());
 }
 
+TEST_F(VersionedLayerClientTest, GetAggregatedData) {
+  olp::client::HRN hrn(kTestHrn);
+
+  constexpr auto kTileId = "23606938";
+  constexpr auto kLayer = "testlayer";
+  constexpr auto kQuadTreeDepth = 4;
+  const auto root_tile = olp::geo::TileKey::FromHereTile(kTileId);
+  const auto tile = root_tile.ChangedLevelTo(16);
+  const auto request = olp::dataservice::read::TileRequest().WithTileKey(tile);
+
+  {
+    SCOPED_TRACE("Requested tile");
+
+    {
+      mock_server_client_->MockAuth();
+      mock_server_client_->MockTimestamp();
+      mock_server_client_->MockGetResponse(
+          mockserver::DefaultResponses::GenerateApisResponse(kTestHrn));
+      mock_server_client_->MockGetResponse(
+          mockserver::DefaultResponses::GenerateVersionResponse(44));
+      mock_server_client_->MockGetResponse(
+          kLayer, root_tile,
+          mockserver::DefaultResponses::GenerateQuadTreeResponse(
+              root_tile, kQuadTreeDepth, {1, 3, 12, 13, 14, 15, 16}));
+      mock_server_client_->MockGetResponse(kLayer, ".*");
```

Comment:
What is this? Is this the blob response? If this is the case should you generate a unique response each time and check that it is the same one you get?

Reply:
Mock server supports regular expressions for expectations. This one is matching any data_handle.
I have added functions to generate data and data_handle by partition, so it can be used while generating responses and to check correctness of loaded data.